### PR TITLE
The `--quiet` option shall silence warnings

### PR DESCRIPTION
**Pull request by happz**
Related to https://github.com/teemtee/tmt/issues/2106

Pull Request Checklist

* [x] implement the feature
* [x] extend the test coverage

### CODE REVIEW

File: tmt/log.py [hunk 1]
```diff
@@ -379,7 +379,7 @@ def filter(self, record: logging.LogRecord) -> bool:
 
 class QuietnessFilter(logging.Filter):
     def filter(self, record: logging.LogRecord) -> bool:
-        if record.levelno not in (logging.DEBUG, logging.INFO):
+        if record.levelno not in (logging.DEBUG, logging.INFO, logging.WARN, logging.WARNING):
```
Comment (lukaszachy) [inline tmt/log.py:382]: Is there reason why we don't compare on value? Eg. `record.levelno > logging.WARNING` ?

**Review by therazix — APPROVED** [review]

File: tests/core/logging/test.sh [hunk 2]
```diff
@@ -26,17 +26,17 @@ rlJournalStart
 
     # TODO: enable the test once --quiet starts silencing all kinds of logging
     # See https://github.com/teemtee/tmt/issues/2106
```
Comment (KwisatzHaderach) [inline tests/core/logging/test.sh:28]: no longer needed, right?

Comment (psss) [reply tests/core/logging/test.sh]: Yeah, makes sense, done in 74870af3.

File: tmt/log.py (hunk 1 above)
Comment (psss) [reply tmt/log.py:382]: Could be shorter, I personally like the explicit listing here. Let's not block on that.

**Review by psss — APPROVED** [review]

**Pull request merged**